tarpc/src/client.rs

```diff
@@ -352,6 +352,7 @@ where
                     let _entered = span.enter();
                     tracing::info!("ReceiveError");
                 }
+                self.pending_requests_mut().close();
```

Comment:
I think there could be pending requests after closing the receiver. Maybe those requests should be completed as well, similar to what's done with in_flight_requests (immediately above)? Something like this: https://github.com/google/tarpc/blob/876fd4724b4ff051d8631be8bbc892023bd98e71/tarpc/src/client/in_flight_requests.rs#L102C19-L102C69

Reply:
I pushed some commits, PTAL.